graph/src/components/store/err.rs

```diff
@@ -50,6 +50,10 @@ pub enum StoreError {
     Poisoned,
     #[error("panic in subgraph writer: {0}")]
     WriterPanic(JoinError),
+    #[error(
+        "found deployment schema version {0} which is not supported. Did you downgrade Graph Node?"
+    )]
+    UnsupportedDeploymentSchemaVersion(i32),
```

Comment:
It might be friendlier if the error message was 
```
deployment '{Qm..}'({sgd123}) uses schema version {N} but this graph-node only supports versions up to {M}. Did you downgrade graph-node?
```

Reply:
I've improved this to show the max version, the deployment id will hopefully show up in the log or higher up in the error context. This required moving `DeploymentSchemaVersion` to the `graph` crate, but that was likely to be required eventually anyways.